crates/red_knot_python_semantic/src/types.rs

```diff
@@ -278,6 +280,10 @@ impl<'db> Type<'db> {
                 Type::Unknown
             }
             Type::BooleanLiteral(_) => Type::Unknown,
+            Type::StringLiteral(_) => {
+                // TODO defer to Type::Instance(<str from typeshed>).member
```

Comment:
This is a somewhat annoying and pedantic nitpick from me, but: this comment isn't _strictly_ accurate. In between `Literal[<some string value>]` and `str`, there's another Python type: [`typing.LiteralString`](https://typing.readthedocs.io/en/latest/spec/literal.html#literalstring). Explaining exactly how `LiteralString` works would take quite a few words and isn't really relevant to this PR 😄 but for now I think I'd just use this as the TODO comment instead:

```suggestion
                // TODO defer to `typing.LiteralString`/`builtins.str`
                // methods from typeshed's stubs
```

Reply:
I guess the relevance to this specific TODO comment is that for many members of a string `Literal`, we can infer the return type as a `LiteralString` instead of `str`? This is true, though it doesn't seem critical to me to reflect that in this TODO comment, though it certainly doesn't hurt. (The other nitpick on the TODO for both this and bytes is that for some methods we can probably safely infer an actual `Literal` type for the member, e.g. we can infer `Literal["foo"].upper()` to be `Literal["FOO"]`)

Reply:
Right, yeah -- as I say, it's kind of a pedantic nitpick. But given the problems that mypy has had with implementing `LiteralString` (well, it still doesn't actually have an implementation, because it requires rewriting a lot of assumptions across the code), I'd prefer for us to keep `LiteralString` in mind from the very beginning when thinking about implementing `str` and `Literal[<some string value>]` types.

Reply:
Will update in a follow-on PR tomorrow – and I am personally a *big* fan of pedantic nitpicks when it comes to stuff like this. 😅